action/protocol/staking/candidate_statemanager.go

```diff
@@ -94,6 +94,56 @@ func (csm *candSM) Upsert(d *Candidate) error {
 	return nil
 }
 
+func (csm *candSM) Commit() error {
+	if err := csm.CandidateCenter.Commit(); err != nil {
+		return err
+	}
+
+	// write update view back to state factory
+	return csm.WriteView(protocolID, csm.CandidateCenter)
+}
+
+func getOrCreateCandCenter(sr protocol.StateReader) (CandidateCenter, error) {
+	c, err := getCandCenter(sr)
+	if err != nil {
+		if errors.Cause(err) == protocol.ErrNoName {
+			// the view does not exist yet, create it
+			cc, err := createCandCenter(sr)
+			return cc, err
+		}
+		return nil, err
```

Comment:
after changing the latest change, do we still need this logic or panic here?

Reply:
i think so, this is getOrCreateCandCenter(), so if get() did not return a view, need to create it.

however, for createGenesis() and Start(), we can use create() for sure

i added a TODO in other 2 places ActiveCandidates() and ReadState(), we can think it thru clearly whether could use create() when pass in sr for those 2

Reply:
only two places use "create", start and read state if not the latest height.